### PR TITLE
feat(financial-aid): File support

### PR DESCRIPTION
**Pull request by addi**
# File support

https://app.asana.com/0/inbox/1108390964729019/1200266597271863/1200571472532425

## What

File upload support for the ósk and display it in veita.

## Why

There is a need.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/api/src/app/modules/application/models/application.model.ts [hunk 1]
```diff
@@ -71,4 +73,7 @@ export class ApplicationModel implements Application {
 
   @Field(() => String)
   readonly state!: ApplicationState
+
+  @Field(() => [ApplicationFileModel], { nullable: true })
```
Comment (addi) [inline apps/financial-aid/api/src/app/modules/application/models/application.model.ts:77]: Should this be null or just empty array?

File: apps/financial-aid/backend/src/app/modules/application/application.service.ts [hunk 2]
```diff
@@ -1,34 +1,58 @@
...
 import { CreateApplicationDto, UpdateApplicationDto } from './dto'
 import { User, Application } from '@island.is/financial-aid/shared'
+import { FileService } from '../file'
 
 @Injectable()
 export class ApplicationService {
   constructor(
     @InjectModel(ApplicationModel)
     private readonly applicationModel: typeof ApplicationModel,
+    private readonly fileService: FileService, // private readonly fileService: FileService,
```
Comment (addi) [inline apps/financial-aid/backend/src/app/modules/application/application.service.ts:18]: What is this comment?

File: apps/financial-aid/backend/src/app/modules/application/application.service.ts [hunk 3]
```diff
@@ -1,34 +1,58 @@
...
-  create(
+  async create(
     application: CreateApplicationDto,
     user: User,
   ): Promise<ApplicationModel> {
-    // this.logger.debug('Creating a new case')
-    return this.applicationModel.create(application)
+    const applModel = await this.applicationModel.create(application)
+
+    //Create fileEvent
```
Comment (addi) [inline apps/financial-aid/backend/src/app/modules/application/application.service.ts:43]: File event?

File: apps/financial-aid/backend/src/app/modules/application/application.service.ts [hunk 4]
```diff
@@ -1,34 +1,58 @@
...
   }
 
-  create(
+  async create(
     application: CreateApplicationDto,
     user: User,
   ): Promise<ApplicationModel> {
-    // this.logger.debug('Creating a new case')
-    return this.applicationModel.create(application)
+    const applModel = await this.applicationModel.create(application)
```
Comment (addi) [inline apps/financial-aid/backend/src/app/modules/application/application.service.ts:41]: applModel is a bad name

File: apps/financial-aid/backend/src/app/modules/file/file.service.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  constructor(
+    @InjectModel(ApplicationFileModel)
+    private readonly fileModel: typeof ApplicationFileModel,
+    private readonly cloudFrontService: CloudFrontService,
+    @Inject(LOGGER_PROVIDER)
+    private readonly logger: Logger,
+  ) {}
+
+  createFile(createFile: CreateFileDto): Promise<ApplicationFileModel> {
+    // this.logger.debug('Creating a new application event')
```
Comment (addi) [inline apps/financial-aid/backend/src/app/modules/file/file.service.ts:25]: Uncomment this log and change it from application event.

File: apps/financial-aid/backend/src/app/modules/file/index.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export { FileModule } from './file.module'
+export { FileService } from './file.service'
+// export { CloudFrontService } from './cloudFront.service'
```
Comment (addi) [inline apps/financial-aid/backend/src/app/modules/file/index.ts:3]: Delete this comment?

File: apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx [hunk 7]
```diff
@@ -84,6 +90,14 @@ const SummaryForm = () => {
   )
 
   const createApplication = async () => {
+    const formatFiles = form?.incomeFiles.map((f) => {
+      return {
+        name: f.name ?? 'mofoJones.png',
```
Comment (addi) [inline apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx:95]: lol, we need to fix this.

File: apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx [hunk 8]
```diff
@@ -364,7 +379,61 @@ const SummaryForm = () => {
...
+                        </Box>
+
+                        <Text>{file.name}</Text>
+                      </a>
+                    )
+                  })}
+                </>
+              )}
+
+              {/* {form?.taxReturnFiles && (
```
Comment (addi) [inline apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx:410]: Need to clean this up.

File: apps/financial-aid/web-osk/src/utils/useFIleUpload.ts [hunk 9]
```diff
@@ -0,0 +1,186 @@
+import { useEffect, useRef, useState } from 'react'
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/useFIleUpload.ts:1]: Fix this filename.

File: apps/financial-aid/web-veita/src/components/Files/Files.tsx [hunk 10]
```diff
@@ -12,10 +12,11 @@ import { useRouter } from 'next/router'
 import * as styles from './Files.treat'
 import cn from 'classnames'
 import { calcDifferenceInDate, getFileType } from '../../utils/formHelper'
+import { ApplicationFile } from '@island.is/financial-aid/shared'
 
 interface Props {
   heading?: string
-  filesArr?: [string]
+  filesArr: ApplicationFile[]
```
Comment (addi) [inline apps/financial-aid/web-veita/src/components/Files/Files.tsx:19]: arrrr, need to fix this.

**Review by MargretFinnboga — APPROVED** [review]
Looks good to me

**Pull request merged**